### ISSUE TITLE
Fwe error and improvements

### ISSUE THREAD

**Issue by oOMrYairOo** [Swizards/QuickBox#49]
Errors:
The commands under root not working
like quickbox returns command not found 
and rest of commands the same error command not found 
from the dashboard when you click enable BTSync 
it tells you wait 5 minutes before enable 
but nothing happens after 15 minutes 
some how it cannot install or enable BTSync via dashboard
from the dashboard if you click on web console it redirects to page 
whit error ''You don't have permission to access /yair.console on this server.''

improvements:

it will be great if you add only https when logging to rutorrent's ui 
disable the http and enable https only for more security
the is no reason to enable http whit https when logging to rutorrent's ui
also if you can add openvpn service to the script 
isp's trolling users who is downloading 24/7 whit vpn isp's can't see shit 
what customers doing whit their connection :)

**Comment by JMSDOnline** [comment]
It sounds as if your script install was corrupted. 

If the alias was not created properly then you would have a white screen returned for the web console. Did you install the script as root. Also, are you working with a server that uses ``/home`` as the primary mount point?

In regards to http(s) - you canot have one without the other. There needs to be a port open (80) in order for traffic to come and go through your server. However; ssl port (443) is indeed forced via .htacess within your QuickBox root as well as ruTorrent. If you are able to access ruTorrent and the QuickBox UI without https, then this most certainly means you have a bad install.

If you could give me a few more details in regards to any errors that may had occurred during install, that would be helpful. 

Additionally, in regards to VPN... yes, this is a feature that will be implemented. In fact, there is already a script for it within the repo ;)

**Comment by oOMrYairOo** [comment]
when i install the quickbox i dont get any error 
everything was smooth during the install 
i try on clean ubuntu 14.04.4 lts server edition and debian 8
also clean install.
the bugs is the same on both linux os's
i can access quickbox and rutorrent on http and https
and yes i run the script as root user 

when i install os i make 10 gb root partition and the rest of disk space as home folder

how can i try the new script whit vpn service?
can you provide git url whit the new script?

**Comment by JMSDOnline** [comment]
> ...i can access quickbox and rutorrent on http and https...

Unfortunately, we cannot reproduce this. There is an .htaccess file with the following contents in the rutorrent/dashboard root directories that enforces otherwise. 

Contents of the .htaccess are as follows:
```
RewriteEngine On
RewriteCond %{HTTPS} !=on
RewriteRule ^/?(.*) https://%{SERVER_NAME}/$1 [R,L]
```

I just ran this test against the Distros/OS you say this bug exist Ubuntu 14 & Debian 8. In both scenerios, I could not access without being directed to port 443 as https. I will continue to reproduce, however, this is the first case I have had someone claim such.

As per the vpn script.

``curl -LO https://raw.githubusercontent.com/Swizards/QuickBox/master/vpn.sh``

Then run with ``bash vpn.sh``

This should install and activate your VPN client as well as package up your certificate and .ovpn file to be accessed at a link the script specifies on completion. This may have some kinks atm as the main focus has been additional feature implementation and stability testing coupled with security trials.

**Comment by oOMrYairOo** [comment]
ok ... so now i make new fresh install of debian 8
i update the os then run the script again as root
now it seems everything i working fine except fwe commands under root
the command deleteSeedBoxUser doesn't work when i type this command under root user 
i get message command not found 
in the dashbord in packages when you click install BTSync it tells you that is going to install BTSync
but nothing happens after that the BTSync stays disabled and when you click the button to enable it 
again nothing happens. it stays disabled.

now new bug ... when i install on fresh debian after everything was installed i get message in rutorrent that the disk space is low the red error so i log in to debian as root and set disk to 60 GB of 80 Gb total 
after i rebooted the server to see if everything was working i got again the error that disk space is low ( the red error ).

it seems that every fresh install of debian different bugs .... 

about the vpn script thanks i try it another time :)

**Comment by JMSDOnline** [comment]
@oOMrYairOo , out of curiosity since I really cannot duplicate any of your issues on the permitted OS/Distro - who is the provider you are using and is this a VPS or a dedicated machine?

**Comment by JMSDOnline** [comment]
For the Installer/Uninstaller for BTSync you can patch this file by doing:

```
nano /usr/bin/installpackage-btsync
```
Change the line:
```
MASTER=$(cat /root/master.txt)
```
to
```
MASTER=$(cat /etc/apache2/master.txt)
```

You will want to do this also for the package uninstaller at ``/usr/bin/removepackage-btsync``

**Comment by oOMrYairOo** [comment]
Thanks for the replay :)
for now im testing this script on virtual box 
long time i searching for good script for full seedbox setup 
so before i rent some dedicated server i testing in virtual environment

**Comment by JMSDOnline** [comment]
Well, in all honesty... I think if you are testing this locally or otherwise via virtual box... this is going to cause some issues with the full placement of the script as it may be limited to the virtual environment it is contained in.

Try spinning up a DigitalOcean droplet and testing there for more live results as to the scripts potential.

As an addendum to this - the #63 BTSync package installer/uninstaller has been fixed.

**Comment by oOMrYairOo** [comment]
in virtualbox is like on hardware instalation 
no kernel changes 
other scripts working fine 

this script is the best i found whit dashbord and quotas without  need webmin
just few things still need some touch 

by the way in your post on torrent invites 
ther are other people talking about the same bugs that i found 
commands under root not all working 
BTSync now that you fix 
and quotas that need to setup on debian and ubuntu os's

**Comment by JMSDOnline** [comment]
I have not encountered any issues with quotas unless this is (as mentioned) on a **/home** mounted install. All commands tested and working without error... again, the number of errors occurring are due to people disregarding the notice about the issue with **/home** mounted partitions.

Virtual Box may be hardware supported, however, drivers are can be/usually are limited as is the network.

PS: I do not focus on third-party sites outside of this repo as I cannot give focus to the various forums due to my daily schedule.

Thank you for your support and suggestions. Any additional concerns you have will be noted and set to a timeline schedule.

:wink:

**Comment by oOMrYairOo** [comment]
:) no problem happy to help and support projects like this ...
as for BTSync: i tryed now the fix that you post ... now its working 
but you cannot access to btsyncs's gui i got messege ''This site can’t be reached
10.10.25.100 refused to connect.''
the full address is on my local network 
10.10.25.100:8888/gui/
for the quotas now i install debian 8 without home partition
and run the script again ...

**Comment by JMSDOnline** [comment]
I'll double-test BTSync as I have a VM spinning up with the install running now. I will further test the root commands to see what issues there may be there. Debian 8.

**Comment by JMSDOnline** [comment]
For your virtual box test... Do you have 2 IP assigned? One being a private IP as well as one public. I have noticed that local virtual box testing will fail in these sorts of scenarios due to the use of a private IP only available on the local network. In this instance one would need to modify your local computers /hosts/ file to be able to read UI assigned to the IP.... especially in the even of ports being used. Local computers firewall have a knack for blocking unassigned ports.

Again, these are just my personal experiences.

**Comment by oOMrYairOo** [comment]
no i have one interface on virtualbox that is bridged to my actual local network

**Comment by oOMrYairOo** [comment]
well i just now finish to install fresh debian 
only whit root partition ...no home partition 
i run the script after reboot  i run ''setdisk'' i added 
all 250GB of 250GB root partition but still after reboot 
i got warning that disk is full when is empty at all.

**Comment by JMSDOnline** [comment]
Depends on how long you are waiting after you reboot. It can take the BTSync daemon up to 5 minutes to fire up after reboot. No reason for it to stop working after a reboot as these changes are made within your ``/home/$user/.startup`` file. If you see ``BTSYNC = Yes`` within that file then there is the possibility it is is simply taking a few minutes to ignite the daemon. Either that or the system cron is borked and not picking up the QuickBox configuration... but this is unlikely.

Out of curiosity, are you rebooting the server immediately after installing the script? This is a necessity as the option at the end of install implies.

**Comment by oOMrYairOo** [comment]
Yes after as script finished installing everything i reboot the server.
